brave/src/test/java/com/linecorp/armeria/server/brave/BraveServiceIntegrationTest.java

```diff
@@ -65,23 +78,63 @@ protected void init() {
             return HttpResponse.of(HttpStatus.NOT_FOUND);
         });
         sb.service("/foo", (ctx, req) -> HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8, "bar"));
-        sb.service("/extra",
+        sb.service("/async", (ctx, req) -> asyncResponse(future ->
+            future.complete(HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8, "bar"))));
+
+        sb.service("/exception", (ctx, req) -> {
+            // TODO: how do we set status 503 and also retain the cause's message?
+            throw new IllegalStateException("not ready");
+        });
+        sb.service("/exceptionAsync", (ctx, req) -> asyncResponse(future ->
+            // TODO: how do we set status 503 and also retain the cause's message?
```

Comment:
I tried status exception with cause, but that eats the causes message in favor of the generic message from the status code.